src/v3/plugins/github/graphView.test.js

```diff
@@ -0,0 +1,383 @@
+          // $ExpectFlowError
+          const badEdge = GE.createEdge.references(userlike, pull);
+          failsForEdge(badEdge);
+        });
+        it("src must be the referrer in edge address", () => {
+          const references = GE.createEdge.references(issue, review);
+          (references: any).src = GN.toRaw(pull);
+          const g = exampleWithParents().addEdge(references);
+          expect(() => new GraphView(g)).toThrow("Invariant: Expected src");
+        });
+        it("dst must be referent in edge address", () => {
+          const references = GE.createEdge.references(issue, review);
+          (references: any).dst = GN.toRaw(pull);
+          const g = exampleWithParents().addEdge(references);
+          expect(() => new GraphView(g)).toThrow("Invariant: Expected dst");
+        });
+      });
+      describe("has parent edges", () => {
+        it("must satisfy specific hom relationships", () => {
+          const g = new Graph()
+            .addNode(GN.toRaw(repo))
+            // $ExpectFlowError
+            .addEdge(GE.createEdge.hasParent(repo, repo));
+          expect(() => new GraphView(g)).toThrow("Invariant: Edge");
+        });
+        it("must be unique", () => {
+          const otherRepo = {type: "REPO", owner: "foo", name: "bar"};
+          const g = exampleWithParents();
+          g.addNode(GN.toRaw(otherRepo));
+          const otherParent = {
+            src: GN.toRaw(issue),
+            dst: GN.toRaw(otherRepo),
+            address: EdgeAddress.append(GE._Prefix.hasParent, "foobar"),
+          };
+          g.addEdge(otherParent);
+          expect(() => new GraphView(g)).toThrow("Parent invariant");
+        });
+        it("must match the parent specified in the node address", () => {
+          const otherRepo = {type: "REPO", owner: "foo", name: "bar"};
+          const otherParent = GE.createEdge.hasParent(issue, otherRepo);
+          const g = new Graph()
+            .addNode(GN.toRaw(issue))
+            .addNode(GN.toRaw(otherRepo))
+            .addEdge(otherParent);
+          expect(() => new GraphView(g)).toThrow("has the wrong parent");
+        });
+        it("must match child specified in the edge address", () => {
+          const parent = GE.createEdge.hasParent(issue, repo);
+          (parent: any).src = GN.toRaw(pull);
+          const g = new Graph()
+            .addNode(GN.toRaw(pull))
+            .addNode(GN.toRaw(repo))
+            .addEdge(parent);
+          expect(() => new GraphView(g)).toThrow("Invariant: Expected src");
+        });
+      });
+    });
+
+    it("are properly re-entrant", () => {
+      const g = new Graph();
+      const view = new GraphView(g);
+      // no error, empty graph is fine
+      view.graph();
+      // introduce an invariant violation (no parent)
+      g.addNode(GN.toRaw(issue));
+      try {
+        view.graph();
+      } catch (_) {}
+      expect(() => view.graph()).toThrow("invariant violated");
```

Comment:
Cool; confirmed that this fails if the `finally` is squashed. Thanks for
writing this.